File: CS/WebSite/App_Code/XpoObject.cs
using System;
using System.Collections.Generic;
using System.Web;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.Xpo.Metadata;

namespace PersistentObjects {
    [Persistent("Arts")]
    public class ArtsEntity : XPCustomObject {
        public ArtsEntity(Session session) : base(session) { }
        DateTime lastWriteTime;
        string name;
        int id;
        int pid;
        bool isFolder;

        public DateTime LastWriteTime {
            get { return lastWriteTime; }
            set { SetPropertyValue("LastWriteTime", ref lastWriteTime, value); }
        }
        public string Name {
            get { return name; }
            set { SetPropertyValue("Name", ref name, value); }
        }
        [Persistent("ID"), Key(AutoGenerate = true)]
        public int Id {
            get { return id; }
            set { SetPropertyValue("Id", ref id, value); }
        }
        [Persistent("ParentID")]
        public int Pid {
            get { return pid; }
            set { SetPropertyValue("Pid", ref pid, value); }
        }
        public bool IsFolder {
            get { return isFolder; }
            set { SetPropertyValue("IsFolder", ref isFolder, value); }
        }
        [Delayed(true)]
        public byte[] Data {
            get { return GetDelayedPropertyValue<byte[]>("Data"); }
            set { SetDelayedPropertyValue<byte[]>("Data", value); }
        }
    }
}

File: VB/WebSite/App_Code/XpoObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Web
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports DevExpress.Xpo.Metadata

Namespace PersistentObjects
	<Persistent("Arts")> _
	Public Class ArtsEntity
		Inherits XPCustomObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private lastWriteTime_Renamed As DateTime
		Private name_Renamed As String
		Private id_Renamed As Integer
		Private pid_Renamed As Integer
		Private isFolder_Renamed As Boolean

		Public Property LastWriteTime() As DateTime
			Get
				Return lastWriteTime_Renamed
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("LastWriteTime", lastWriteTime_Renamed, value)
			End Set
		End Property
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
		<Persistent("ID"), Key(AutoGenerate := True)> _
		Public Property Id() As Integer
			Get
				Return id_Renamed
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Id", id_Renamed, value)
			End Set
		End Property
		<Persistent("ParentID")> _
		Public Property Pid() As Integer
			Get
				Return pid_Renamed
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Pid", pid_Renamed, value)
			End Set
		End Property
		Public Property IsFolder() As Boolean
			Get
				Return isFolder_Renamed
			End Get
			Set(ByVal value As Boolean)
				SetPropertyValue("IsFolder", isFolder_Renamed, value)
			End Set
		End Property
		<Delayed(True)> _
		Public Property Data() As Byte()
			Get
				Return GetDelayedPropertyValue(Of Byte())("Data")
			End Get
			Set(ByVal value As Byte())
				SetDelayedPropertyValue(Of Byte())("Data", value)
			End Set
		End Property
	End Class
End Namespace
